### PR TITLE
feat: Enhanced filters for argo {watch,get,submit}

### LINKED ISSUE #2428: give argo watch and argo submit --watch the option to hide skipped steps
# Summary

`argo watch` gets cluttered if there are too many false conditionals. There should be an option to mute the lines of those steps.

# Motivation

`argo watch` is a great way to look at the full workflow as it's running. However, it tends to get quite cluttered when there are multiple routes that a workflow can go down, controlled by `when` conditionals. For instance, an exit handler might take a form like [this](https://github.com/argoproj/argo/blob/master/examples/exit-handlers.yaml) with several different possible outcomes. Mine looks this complicated:
```
 ✔ onExit (exit-handler)
 ├-·-○ 
 | └-✔ 
 |   └-·-○ 
 |     ├-○ 
 |     ├-✔ 
 |     | └---✔ 
 |     └-○ 
 ├---○ 
 └---○ 
```
For complicated workflows like these, it would be nice to have the option to "mute" these steps that aren't running, so I'd just see this instead:
```
 ✔ onExit (exit-handler)
 └---✔ 
     └---✔ 
         └---✔ 
```

Beyond exit handlers, adding conditionals can be very helpful for various checks on parameters, like `assert` statements. I've actually shied away from including a lot of these because of how it clouds `argo watch`, so it would be nice if I didn't have to worry about that.

# Proposal

For a user's perspective, a `--mute-skipped` (or similar, I haven't thought through the naming) flag would be added to both `argo watch` and `argo submit --watch`.

---

**Message from the maintainers**:

If you wish to see this enhancement implemented please add a 👍 reaction to this issue! We often sort issues this way to know what to prioritize.

### PR DESCRIPTION
**Pull request by simster7**
Closes #2428 

Adds `--node-field-selector` filter to `argo {watch,get,submit}`.

### CODE REVIEW

**Comment by alexec** [comment]
@simster7 @sarabala1979 this appears to have been abandoned?

**Comment by simster7** [comment]
Thanks @alexec. I have merged and resolved conflicts. Can either of you review please? @sarabala1979

**Comment by simster7** [comment]
Sure, good suggestion

**Comment by simster7** [comment]
> Sure, good suggestion

Done, ready for another review

File: cmd/argo/commands/submit.go [hunk 1]
```diff
@@ -76,6 +81,8 @@ func NewSubmitCommand() *cobra.Command {
 	command.Flags().StringVarP(&submitOpts.ParameterFile, "parameter-file", "f", "", "pass a file containing all input parameters")
 	command.Flags().StringVarP(&submitOpts.Labels, "labels", "l", "", "Comma separated labels to apply to the workflow. Will override previous values.")
 	command.Flags().StringVar(&from, "from", "", "Submit from an existing `kind/name` E.g., --from=cronwf/hello-world-cwf")
+	command.Flags().StringVar(&cliSubmitOpts.getArgs.status, "status", "", "Filter by status (Pending, Running, Succeeded, Skipped, Failed, Error). Should only be used with --watch.")
```
Comment (alexec) [inline cmd/argo/commands/submit.go:84]: do you think we should have short hands for these?

File: cmd/argo/commands/watch.go [hunk 2]
```diff
@@ -60,17 +66,20 @@ func watchWorkflow(wfName string) {
 		}
 		errors.CheckError(err)
 		wf := event.Object
-		printWorkflowStatus(wf)
+		if wf == nil {
```
Comment (alexec) [inline cmd/argo/commands/watch.go:69]: I'm pretty sure nil means error?

File: cmd/argo/commands/get.go [hunk 3]
```diff
@@ -443,8 +446,21 @@ func renderChild(w *tabwriter.Writer, wf *wfv1.Workflow, nInfo renderNode, depth
 
 // Main method to print information of node in get
 func printNode(w *tabwriter.Writer, node wfv1.NodeStatus, nodePrefix string, getArgs getFlags) {
-	if getArgs.status != "" && string(node.Phase) != getArgs.status {
-		return
+	if getArgs.status != "" {
```
Comment (alexec) [inline cmd/argo/commands/get.go:449]: can we move this code into a single func

```
func (a getArgs) GetNodeFieldSelector() (fields.Selector, error)
```

Then you can unit test that function easily.

File: cmd/argo/commands/watch.go (hunk 2 above)
Comment (simster7) [reply cmd/argo/commands/watch.go:69]: Correct: if the `wf` does not exists we don't want to print it.

File: cmd/argo/commands/get.go (hunk 3 above)
Comment (simster7) [reply cmd/argo/commands/get.go:449]: Sure thing

File: cmd/argo/commands/submit.go (hunk 1 above)
Comment (simster7) [reply cmd/argo/commands/submit.go:84]: Would rather not add it proactively and wait to see if they are requested by the community `kubectl --field-selector` does not have a shorthand.

**Pull request merged**